pageserver/src/layered_repository/storage_layer.rs

```diff
@@ -125,4 +128,9 @@ pub trait Layer: Send + Sync {
     fn get_seg_size(&self, lsn: Lsn) -> Result<u32>;
 
     fn get_seg_exists(&self, lsn: Lsn) -> Result<bool>;
+
+    fn is_incremental(&self) -> bool;
+
+    fn unload(&self) -> Result<()>;
```

Comment:
To me, it seems quite strange that the trait has an `unload` function, while it doesn't have a `load`.
Though, I understand that it is called implicitly on demand from some other functions. Let's explain it in the comments for `get_seg_size()` and `get_page_reconstruct_data()`

This trait needs more comments, especially for `is_dropped`, `unload` and `delete` functions.

Reply:
Added comments